code/modules/mob/living/simple_animal/hostile/retaliate/king_of_goats.dm

```diff
@@ -0,0 +1,328 @@
+
+
+/mob/living/simple_animal/hostile/retaliate/goat/king/phase2/update_icon()
+	var/matrix/M = new
+	if(phase3)
+		icon_state = "king_goat3"
+		icon_living = "king_goat3"
+		M.Scale(1.5)
+	else
+		M.Scale(1.25)
+	transform = M
+	pixel_y = 10
+
+/mob/living/simple_animal/hostile/retaliate/goat/king/phase2/Life()
+	. = ..()
+	if(stat != DEAD)
+		var/sound/song_played = sound(current_song)
+
+		for(var/mob/M in range(10, src))
+			if(!M.client || !(M.client.prefs.toggles & SOUND_INSTRUMENTS))
+				continue
+			if(!(M in rangers) || world.time > rangers[M])
+				M.stop_sound_channel(CHANNEL_JUKEBOX)
+				rangers[M] = world.time + current_song_length
+				M.playsound_local(null, null, 30, channel = CHANNEL_JUKEBOX, S = song_played)
+		for(var/mob/L in rangers)
+			if(get_dist(src, L) > 10)
+				rangers -= L
+				if(!L || !L.client)
+					continue
+				L.stop_sound_channel(CHANNEL_JUKEBOX)
+	else
+		for(var/mob/L in rangers)
+			rangers -= L
+			if(!L || !L.client)
+				continue
+			L.stop_sound_channel(CHANNEL_JUKEBOX)
+	if(move_to_delay < 3)
+		move_to_delay += 0.1
+	if((health <= 150 && !phase3 && spellscast == 5) || (stat == DEAD && !phase3)) //begin phase 3, reset spell limit and heal
+		phase3_transition()
+	if(!.)
+		return FALSE
+	if(special_attacks >= 6 && melee_damage_type != BRUTE)
+		visible_message("<span class='cult'>The energy surrounding \the [src]'s horns dissipates.</span>")
+		melee_damage_type = BRUTE
+
+/mob/living/simple_animal/hostile/retaliate/goat/king/proc/OnDeath()
+	visible_message("<span class='cult'>\The [src] lets loose a terrific wail as its wounds close shut with a flash of light, and its eyes glow even brighter than before!</span>")
+	var/mob/living/simple_animal/hostile/retaliate/goat/king/phase2/nextgoat = new(src.loc)
+	nextgoat.enemies |= enemies
+	qdel(src);
+
+/mob/living/simple_animal/hostile/retaliate/goat/king/phase2/OnDeath()
+	for(var/mob/L in rangers)
+		rangers -= L
+		if(!L || !L.client)
+			continue
+		L.stop_sound_channel(CHANNEL_JUKEBOX)
+	if(phase3)
+		visible_message("<span class='cult'>\The [src] shrieks as the seal on his power breaks and he starts to break apart!</span>")
+		new /obj/structure/ladder/unbreakable/goat(loc)
+		new /obj/item/gun/energy/goatgun(loc)
+		new /obj/item/toy/plush/goatplushie/angry/kinggoat(loc) //If someone dies from this after beating the king goat im going to laugh
+
+/mob/living/simple_animal/hostile/retaliate/goat/king/death()
+	..()
+	OnDeath()
+
+/mob/living/simple_animal/hostile/retaliate/goat/king/phase2/Destroy()
+	for(var/mob/L in rangers)
+		rangers -= L
+		if(!L || !L.client)
+			continue
+		L.stop_sound_channel(CHANNEL_JUKEBOX)
+	. = ..()
+
+/mob/living/simple_animal/hostile/retaliate/goat/king/AttackingTarget()
+	. = ..()
+	if(isliving(target))
+		var/mob/living/L = target
+		if(L.stat == DEAD)
+			L.gib()
+		if(prob(stun_chance))
+			L.Paralyze(5)
+			L.confused += 1
```

Comment:
adding 1 to confused is going to do basically nothing

also, confusing someone in a boss fight is kind of a lame mechanic, since it's gonna be RNG whether or not they can actually run away from/kite the boss or if their character decides to die by running back towards the boss (or to the side)